gateway/src/user/user.service.ts

```diff
@@ -0,0 +1,29 @@
+import { Inject, Injectable } from '@nestjs/common';
+import { ClientProxy } from '@nestjs/microservices';
+import { lastValueFrom, Observable } from 'rxjs';
+import { UserSignUpRequestBody } from './dto/user-reqbody.dto';
+import { UserSignUpEventMsg } from '../../../lib/user/event-msg.dto';
+
+@Injectable()
+export class UserService {
+  constructor(
+    @Inject('USER_SERVICE') private readonly userClient: ClientProxy,
+  ) {}
+
+  signup(userSignUpRequestBody: UserSignUpRequestBody): Observable<any> {
```

Comment:
I think using `observable` in service will block HTTP exceptions that return from micro-service, I am using `Promise` and solve issues. Can you test and share result?